src/middlewared/middlewared/plugins/kubernetes_linux/update.py

```diff
@@ -132,6 +132,16 @@ async def validate_data(self, data, schema, old_data):
                         f'{schema}.migrate_applications',
                         f'{applications_ds_name(old_data["pool"])!r} does not exist, migration not possible.'
                     )
+                if not verrors and (
+                    await self.middleware.call(
+                        'pool.dataset.get_instance', old_data['pool'], {'extra': {'retrieve_children': False}}
+                    )
+                )['encrypted']:
+                    verrors.add(
+                        f'{schema}.migrate_applications',
+                        f'Source {old_data["pool"]!r} is encrypted and it is not supported '
+                        'migrating encrypted applications data'
```

Comment:
Instead we can try to replicate it to the new dataset without replicating any of the dataset attributes. In my experience this scrubbed away all potential issues with encryption.

However, afterwards we do need to execute a flow to 
- have the PVC's and some other datasets set to `legacy` mountpath.
- Scrub unwanted parent dataset options comming from the target parent dataset